apps/mobile/src/features/balances/stacks/use-get-stacks-addresses.ts

```diff
@@ -8,8 +8,10 @@ export function useGetStacksAddresses(accountId?: AccountId) {
     return signers.list.map(signer => signer.address);
   }
 
-  const { fingerprint } = accountId;
+  const { accountIndex, fingerprint } = accountId;
   return signers.list
-    .filter(signer => signer.descriptor.includes(fingerprint))
+    .filter(
+      signer => signer.descriptor.includes(fingerprint) && signer.accountIndex === accountIndex
+    )
```

Comment:
It looks like you're repeating logic here, `useStacksSigners` has a `.fromAccountIndex` method to do the filtering?

Reply:
Good catch 👍